packages/aztec.js/src/proof/bilateralSwap/index.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * Constructs AZTEC bilateral swap zero-knowledge proofs
+ *
+ * @module proof
+ */
+const BN = require('bn.js');
+const { padLeft } = require('web3-utils');
+const Keccak = require('../../keccak');
+const bn128 = require('../../bn128');
+
+const helpers = require('./helpers');
+
+const { groupReduction } = bn128;
+
+const bilateralSwap = {};
+bilateralSwap.helpers = helpers;
+
+/**
+ * Construct AZTEC bilateral swap proof transcript
+ *
+ * @method constructProof
+ * @param {Array[Note]} notes array of AZTEC notes
+ * @returns {{proofData:Array[string]}, {challenge: string}} - proof data and challenge
+ */
+bilateralSwap.constructBilateralSwap = (notes, sender) => {
+    // finalHash is used to create final proof challenge
+    const finalHash = new Keccak();
+
+    finalHash.appendBN(new BN(sender.slice(2), 16));
+
+    notes.forEach((note) => {
+        finalHash.append(note.gamma);
+        finalHash.append(note.sigma);
+    });
+
+    const bkArray = [];
+    const blindingFactors = notes.map((note, i) => {
```

Comment:
This looks like the same code as `helpers.getBlindingFactorsAndChallenge`, we should re-use that method if this is the case.

Reply:
They do slightly different things, although they are similar. Inclined to leave as is